core/src/main/java/com/sequenceiq/cloudbreak/service/stack/DefaultStackService.java

```diff
@@ -143,12 +144,14 @@ public Stack create(CbUser user, Stack stack) {
         Stack savedStack = null;
         stack.setOwner(user.getUserId());
         stack.setAccount(user.getAccount());
+        MDCBuilder.buildMdcContext(stack);
         String result = provisionSetups.get(stack.cloudPlatform()).preProvisionCheck(stack);
         if (null != result) {
             throw new BadRequestException(result);
         } else {
             try {
                 savedStack = stackRepository.save(stack);
+                MDCBuilder.buildMdcContext(savedStack);
```

Comment:
The line was inserted here, because this is the first time, where all the necessary fields are set on stack for logging. Without this couple of log messages won't be proper.